extendr-api/src/optional/either.rs

```diff
@@ -0,0 +1,43 @@
+use crate::{Error, Robj};
+use either::Either::{self, Left, Right};
+
+impl<'a, TLeft, TRight> TryFrom<&'a Robj> for Either<TLeft, TRight>
```

Comment:
Maybe this call them `Left` and `Right`?

Reply:
`Either` uses `T` and `R` for their names, so I decided to go the same way.

Reply:
![Screenshot_20230225-144432.png](https://user-images.githubusercontent.com/1063624/221360242-7a2c9195-f351-4482-8c49-8c7ea34f6672.png)

Either use left and right, or R and L, but this current one is a little ugly. Even T and R is better.